zio-logback-http4s/src/main/scala/com/softwaremill/correlator/CorrelationId.scala

```diff
@@ -0,0 +1,99 @@
+package com.softwaremill.correlator
+
+import java.{util => ju}
+
+import cats.data.{Kleisli, OptionT}
+import ch.qos.logback.classic.util.LogbackMDCAdapter
+import org.http4s.util.CaseInsensitiveString
+import org.http4s.{HttpRoutes, Request, Response}
+import org.slf4j.{Logger, LoggerFactory, MDC}
+import zio.{FiberRef, Runtime, Task, UIO, ZIO}
+
+import scala.util.Random
+
+/**
+  * Correlation id support. The `init()` method should be called when the application starts.
+  * See [[https://blog.softwaremill.com/correlation-ids-in-scala-using-monix-3aa11783db81]] for details.
+  */
+final class CorrelationId(
+    val headerName: String = "X-Correlation-ID",
+    logStartRequest: (String, Request[Task]) => Task[Unit] = (cid, req) =>
+      Task(CorrelationId.logger.debug(s"Starting request with id: $cid, to: ${req.uri.path}"))
+)(implicit runtime: Runtime[Any]) {
+  private val MdcKey = "cid"
+  private val random = new Random()
+
+  def init: UIO[Unit] = ZioMDCAdapter.init(runtime)
+
+  def get: UIO[Option[String]] = Task(Option(MDC.get(MdcKey))).orElse(UIO.apply(None))
```

Comment:
Should we prefer to return the `Task` or, as I've done it. transform it as a `UIO`?

Reply:
Hm, well if it can't fail with an exception ... I suppose it shold be a `UIO`?

Reply:
`MDC.get` can throw. In practice, I'm not sure the case where it throws can happen here. I'll stay with a `Task`, just in case.

Reply:
ok :)